spec/requests/api/users_spec.rb

```diff
@@ -51,11 +43,138 @@
     end
 
     context 'アクセストークンがない場合' do
+      it '400でエラーメッセージを出力して失敗する' do
+        expect(subject).to be_bad_request
+        expect(subject.parsed_body).to have_key('message')
+      end
+    end
+  end
+
+  describe 'POST /api/users' do
+    subject do
+      post '/api/users', headers:, params: { name:, email:, password: }
+      response
+    end
+
+    let(:name) { 'hogehgoe' }
+    let(:email) { 'hoge@example.com' }
+    let(:password) { 'foobar' }
+
+    # rubocop:disable RSpec/MultipleMemoizedHelpers
+    context 'アクセストークンが有効の場合' do
+      let!(:user) { create(:user, :admin) }
+      let(:access_token) { AccessToken.new(email: user.email).encode }
+      let(:headers) { { 'Authorization' => "Bearer #{access_token}" } }
+
+      context 'パラメータが適切な場合' do
+        context 'ユーザが存在する場合' do
+          let(:email) { user.email }
+
+          it '422が返って、エラーメッセージを返すこと' do
+            expect(subject).to have_http_status(:unprocessable_entity)
+            expect(subject).to have_error 'email', 'has already been taken'
+            expect { subject }.not_to change(User, :count)
+          end
+        end
+
+        context 'ユーザが存在しない場合' do
+          it '201が返って、作成したユーザを返すこと' do
+            expect(subject).to be_created
+            expect(subject.parsed_body.symbolize_keys).to include(
```

Comment:
ここは symbolize_keys をせずに書けませんっけ。
```
```suggestion
            expect(subject.parsed_body).to include("id", "name", "admin", ...)
```
とか
```suggestion
            expect(subject.parsed_body).to include(*%w[id name admin ...])
```
とかって書き方ができそうです。